cypress/page/personal-form.page.ts

```diff
@@ -0,0 +1,79 @@
+import {Information, FormFields}
+  from "../test_objects/objects-types";
+
+class PersonalFormPage {
+  private personalFormPageURL: string;
+  private calendarBtn: string;
+  private submitBtn: string;
+  private formFields: FormFields;
+
+  constructor() {
+    this.personalFormPageURL = "https://demoqa.com/automation-practice-form/";
+    this.calendarBtn = "#dateOfBirthInput";
+    this.submitBtn = "#submit";
+    this.formFields = {
+      nameInput: "#firstName",
+      lastNameInput: "#lastName",
+      emailInput: "#userEmail",
+      genderInput: ".custom-radio",
+      dateInput: {
+        month: ".react-datepicker__month-select",
+        year: ".react-datepicker__year-select",
+        day: ".react-datepicker__day--0",
+      },
+      mobileNumberInput: "#userNumber",
+      hobbiesInput: ".custom-checkbox",
+      addresInput: "#currentAddress",
+    };
+  }
+
+  private selectGender(info: Information): void {
+    cy.get(this.formFields.genderInput)
+        .filter(`:contains("${info.gender}")`)
+        .click();
+  }
+
+  private selectDate(info: Information): void {
+    const excludeDayOutOfMonth: string = ".react-datepicker__day--outside-month";
+    const dateOfBirth = new Date(info.dateOfBirth);
+
+    cy.get(this.calendarBtn).click().then(() => {
+      cy.get(this.formFields.dateInput.month)
+          .select(`${dateOfBirth.getMonth()}`);
+      cy.get(this.formFields.dateInput.year).select(`${dateOfBirth.getFullYear()}`);
+      if (`${dateOfBirth.getDate()}`.length > 1) {
+        cy.get(`${this.formFields.dateInput.day}${dateOfBirth.getDate()}`)
+            .not(excludeDayOutOfMonth).click();
+      } else {
+        cy.get(`${this.formFields.dateInput.day}0${dateOfBirth.getDate()}`)
+            .not(excludeDayOutOfMonth).click();
+      }
+    });
```

Comment:
this is kindof hard to read and understand. is this the only way to select a date on the datepicker?

Reply:
supposing that I don't want to write the date directly in the field I can't think of another way, because if I tell it to bring the 9th of July there are two occurrences in the calendar, and it brings me the first one that corresponds to the 9th of June. Any suggestions? The same applies to other dates outside the current month.

Reply:
I have already changed it. I think the new logic is a bit better. First I exclude the days outside of the month and then I make the selection.

Reply:
congrats, really good job